frontend/src/modules/EditZing/Components/FilterSidebar.tsx

```diff
@@ -0,0 +1,128 @@
+import { useState, ChangeEvent } from 'react'
+import { FilterSidebarProps } from 'EditZing/Types/ComponentProps'
+import {
+  FiltersContainer,
+  FilterContainer,
+  FilterHeading,
+  FilterHeadingContainer,
+  upSvg,
+  downSvg,
+  SvgContainer,
+} from 'EditZing/Styles/FilterSidebar.style'
+import FormControl from '@mui/material/FormControl'
+import FormGroup from '@mui/material/FormGroup'
+import FormControlLabel from '@mui/material/FormControlLabel'
+import Checkbox from '@mui/material/Checkbox'
+
+export const FilterSidebar = ({
+  filterData,
+  filtersSelected,
+  setFiltersSelected,
+}: FilterSidebarProps) => {
+  const [sectionsCollapsed, setSectionsCollapsed] = useState<boolean[]>(
+    Array(Object.keys(filterData).length).fill(false)
+  )
+
+  const isSectionCollapsed = (index: number) => {
+    return sectionsCollapsed[index]
+  }
+
+  const checkboxComponent = (optionStr: string, category: string) => {
+    const handleCheckboxChange = (event: ChangeEvent<HTMLInputElement>) => {
+      if (event.target.checked) {
+        // add new category to filtersSelected if category doesn't exist
+        if (!filtersSelected[category]) {
+          setFiltersSelected({ ...filtersSelected, [category]: [optionStr] })
+        }
+        // add option to existing category
+        else {
+          setFiltersSelected({
+            ...filtersSelected,
+            [category]: [...filtersSelected[category], optionStr],
+          })
+        }
+      } else {
+        // filter out option from its category in filtersSelected object
+        let newFiltersSelected = {
+          ...filtersSelected,
+          [category]: filtersSelected[category].filter(
+            (option) => option !== optionStr
+          ),
+        }
+        // delete category if it is now empty
+        if (newFiltersSelected[category].length === 0) {
+          delete newFiltersSelected[category]
+        }
+        setFiltersSelected(newFiltersSelected)
+      }
+    }
+    return (
+      <FormControlLabel
+        key={optionStr + '-filtercheckbox'}
+        control={
+          <Checkbox
+            onChange={handleCheckboxChange}
+            name={optionStr}
+            value={optionStr}
+            color="primary"
+          />
+        }
+        label={optionStr}
+      />
+    )
+  }
+
+  const filterCheckboxes = (key: string, index: number, category: string) => {
```

Comment:
what's the style pref here? should a function that returns a JSX element be camel case with first letter caps? or is this fine

Reply:
I think either is acceptable, it really depends on how it's used. E.g. is the function returning an actual "element" or is it just helping choose what elements to display? Note that you can make literal JSX elements if you put the function arguments in an object (in this case the location of `key` might have to be moved):
```js
const CheckboxComponent = ({ optionStr, category }: { optionStr: string, category: string }) => {...}

<CheckboxComponent optionStr={option.description} category={category} />
```